src/codegen/tuple.cpp

```diff
@@ -0,0 +1,60 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// tuple.cpp
+//
+// Identification: src/codegen/tuple.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "codegen/tuple.h"
+
+#include "catalog/schema.h"
+#include "codegen/proxy/data_table_proxy.h"
+#include "codegen/proxy/tuple_runtime_proxy.h"
+#include "codegen/type/sql_type.h"
+#include "storage/data_table.h"
+
+namespace peloton {
+namespace codegen {
+
+Tuple::Tuple(storage::DataTable &table): table_(table) {}
```

Comment:
Change the tuple constructor to accept the `catalog::Schema` instead of the data table.